packages/monaco/src/browser/monaco-init.ts

```diff
@@ -50,6 +50,56 @@ import { MonacoQuickInputImplementation } from './monaco-quick-input-service';
 import { IQuickInputService } from '@theia/monaco-editor-core/esm/vs/platform/quickinput/common/quickInput';
 import { IStandaloneThemeService } from '@theia/monaco-editor-core/esm/vs/editor/standalone/common/standaloneTheme';
 import { MonacoStandaloneThemeService } from './monaco-standalone-theme-service';
+import { ContentHoverWidget } from '@theia/monaco-editor-core/esm/vs/editor/contrib/hover/browser/contentHoverWidget';
+import { IPosition } from '@theia/monaco-editor-core/esm/vs/editor/common/core/position';
+
+// https://github.com/microsoft/vscode/blob/1430e1845cbf5ec29a2fc265f12c7fb5c3d685c3/src/vs/editor/contrib/hover/browser/resizableContentWidget.ts#L13-L14
+const VSCODE_TOP_HEIGHT = 30;
+const VSCODE_BOTTOM_HEIGHT = 24;
+
+// VS Code uses 30 pixel for top height, and 24 pixels for bottom height, but Theia uses 32 pixel for the top and 22 for the bottom.
+// https://github.com/eclipse-theia/theia/blob/b752ea690bdc4e7c5d9ab98a138504ead05be0d1/packages/core/src/browser/style/menus.css#L22
+// https://github.com/eclipse-theia/theia/blob/b752ea690bdc4e7c5d9ab98a138504ead05be0d1/packages/core/src/browser/style/status-bar.css#L18
+// https://github.com/eclipse-theia/theia/issues/14826
+function patchContentHoverWidget(topPanelHeight = 32): { setActualTopHeightForContentHoverWidget: (value: number) => void } {
+    let _actualTopHeight = topPanelHeight;
+    function getTopHeightDiff(): number {
+        return _actualTopHeight - VSCODE_TOP_HEIGHT;
+    }
+
+    const actualBottomHeight = 22; // Theia's status bar height
```

Comment:
maybe we can move this to an arg with default value export the function, and if downstream projects redefine the status bar height, they can reuse this code as-is without further prototype patching

Reply:
You mean like we do for `actualTopHeight`? That seems reasonable.

Reply:
Please see the new API proposal.